File: samples/da-trey-research-python/src/services/utilities.py
class HttpError(Exception):
    """Custom HTTP error exception."""

    def __init__(self, status: int, message: str):
        super().__init__(message)
        self.status = status
        self.message = message


def clean_up_parameter(name: str, value: str) -> str:
    """Clean up common issues with Copilot parameters."""
    val = value.lower()

    if "trey" in val or "research" in val:
        new_val = val.replace("trey", "").replace("research", "").strip()
        print(f"   ❗ Plugin name detected in the {name} parameter '{val}'; replacing with '{new_val}'.")
        val = new_val

    if val == "<user_name>":
        print(f"   ❗ Invalid name '{val}'; replacing with 'avery'.")
        val = "avery"

    if name == "role" and val == "consultant":
        print(f"   ❗ Invalid role name '{val}'; replacing with ''.")
        val = ""

    if val == "null":
        print(f"   ❗ Invalid value '{val}'; replacing with ''.")
        val = ""

    return val

File: samples/da-trey-research-auth-csharp/TreyResearchApi/Services/Utilities.cs
namespace TreyResearchApi.Services;

public class HttpError : Exception
{
    public int Status { get; }

    public HttpError(int status, string message) : base(message)
    {
        Status = status;
    }
}

public static class Utilities
{
    public static string CleanUpParameter(string name, string value)
    {
        var val = value.ToLowerInvariant();

        if (val.Contains("trey") || val.Contains("research"))
        {
            var newVal = val.Replace("trey", "").Replace("research", "").Trim();
            Console.WriteLine($"   ❗ Plugin name detected in the {name} parameter '{val}'; replacing with '{newVal}'.");
            val = newVal;
        }

        if (val == "<user_name>")
        {
            Console.WriteLine($"   ❗ Invalid name '{val}'; replacing with 'avery'.");
            val = "avery";
        }

        if (name == "role" && val == "consultant")
        {
            Console.WriteLine($"   ❗ Invalid role name '{val}'; replacing with ''.");
            val = "";
        }

        if (val == "null")
        {
            Console.WriteLine($"   ❗ Invalid value '{val}'; replacing with ''.");
            val = "";
        }

        return val;
    }
}
